### PR TITLE
Fix: scope for TSImportEqualsDeclaration

### PR DESCRIPTION
**Pull request by armano2**
Import equal should declare "ImportBinding" variable in scope.

example:
```ts
import foo = require('bar')
```

https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require

this PR fixes ticket #555

### CODE REVIEW

File: tests/lib/__snapshots__/scope-analysis.js.snap [hunk 1]
```diff
@@ -4480,6 +4480,101 @@ Object {
...
+              "name": "foo",
+              "range": Array [
+                7,
+                10,
+              ],
+              "type": "Identifier",
+            },
+          ],
+          "name": "foo",
+          "references": Array [],
```
Comment (mysticatea) [inline tests/lib/__snapshots__/scope-analysis.js.snap]: In TS, isn't the `require` a reference?

Comment (j-f1) [reply tests/lib/__snapshots__/scope-analysis.js.snap]: No — `import foo = require('bar')` is different from `const foo = require('bar')

Comment (armano2) [reply tests/lib/__snapshots__/scope-analysis.js.snap]: i will investigate that

Comment (armano2) [reply tests/lib/__snapshots__/scope-analysis.js.snap]: @mysticatea  `import foo = require('bar')` is different syntax to write `import foo from 'bar'`
and `require` is not reference like `from` is not.

typescript have this weird syntax to distinct `export =` from `export default`

Comment (mysticatea) [reply tests/lib/__snapshots__/scope-analysis.js.snap]: Thank you for the explanation! I got it.

**Pull request merged**